kernel/src/Boot/Patches/Patches672.cpp

```diff
@@ -123,39 +124,105 @@ void Mira::Boot::Patches::install_prerunPatches_672()
 	kmem = (uint8_t *)&gKernelBase[0x0010F879];
 	kmem[0] = 0xEB;
 
-	// Enable debug rif's
-	kmem = (uint8_t*)&gKernelBase[0x66AEB0];
-	kmem[0] = 0xB0;
-	kmem[1] = 0x01;
-	kmem[2] = 0xC3;
+	// second ptrace patch
+	// via DeathRGH
+	kmem = (uint8_t *)&gKernelBase[0x0010FD22];
+	kmem[0] = 0xE9;
+	kmem[1] = 0xE2;
+	kmem[2] = 0x02;
+	kmem[3] = 0x00;
+	kmem[4] = 0x00;
+
+	// setlogin patch (for autolaunch check)
+	kmem = (uint8_t *)&gKernelBase[0x0010EC1C];
+	kmem[0] = 0x48;
+	kmem[1] = 0x31;
+	kmem[2] = 0xC0;
 	kmem[3] = 0x90;
+	kmem[4] = 0x90;
 
-	// Enable debug rifs 2
-	kmem = (uint8_t*)&gKernelBase[0x66AEE0];
-	kmem[0] = 0xB0;
-	kmem[1] = 0x01;
-	kmem[2] = 0xC3;
+	// Patch to remove vm_fault: fault on nofault entry, addr %llx
+	kmem = (uint8_t *)&gKernelBase[0x000BC8F6];
+	kmem[0] = 0x90;
+	kmem[1] = 0x90;
+	kmem[2] = 0x90;
 	kmem[3] = 0x90;
+	kmem[4] = 0x90;
+	kmem[5] = 0x90;
+
+	// patch mprotect to allow RWX (mprotect) mapping 6.72
+	kmem = (uint8_t *)&gKernelBase[0x00451DB8];
+	kmem[0] = 0x90;
+	kmem[1] = 0x90;
+	kmem[2] = 0x90;
+	kmem[3] = 0x90;
+	kmem[4] = 0x90;
+	kmem[5] = 0x90;
 
-	// Disable pfs checks
-	kmem = (uint8_t*)&gKernelBase[0x6A8EB0];
+	// flatz disable pfs signature check
+	kmem = (uint8_t *)&gKernelBase[0x006A8EB0];
 	kmem[0] = 0x31;
 	kmem[1] = 0xC0;
 	kmem[2] = 0xC3;
-	kmem[3] = 0x90;
 
-	kmem = (uint8_t*)kdlsym(global_settings_base);
-	kmem[0x36] |= 0x14;
-	kmem[0x59] |= 0x01; // sceSblRcMgrIsAllowULDebugger
-	kmem[0x59] |= 0x02; // sceSblRcMgrIsAllowSLDebugger
-	kmem[0x5A] |= 0x01;
-	kmem[0x78] |= 0x01;
+	// flatz enable debug RIFs
+	kmem = (uint8_t *)&gKernelBase[0x0066AEB0];
+	kmem[0] = 0xB0;
+	kmem[1] = 0x01;
+	kmem[2] = 0xC3;
+
+	kmem = (uint8_t *)&gKernelBase[0x0066AEE0];
+	kmem[0] = 0xB0;
+	kmem[1] = 0x01;
+	kmem[2] = 0xC3;
 
 	// Enable *all* debugging logs (in vprintf)
 	// Patch by: SiSTRo
-	kmem = (uint8_t*)&gKernelBase[0x00123367];
-	kmem[0] = 0xEB; // jmp +0x3D
+	kmem = (uint8_t *)&gKernelBase[0x00123367];
+	kmem[0] = 0xEB;
 	kmem[1] = 0x3B;
 
+	// flatz allow mangled symbol in dynlib_do_dlsym
+	kmem = (uint8_t *)&gKernelBase[0x00417A27];
+	kmem[0] = 0x90;
+	kmem[1] = 0x90;
+	kmem[2] = 0x90;
+	kmem[3] = 0x90;
+	kmem[4] = 0x90;
+	kmem[5] = 0x90;
+
+	// Enable mount for unprivileged user
+	kmem = (uint8_t *)&gKernelBase[0x0044026A];
```

Comment:
What is this change required for?

Reply:
This has been included in the previous firmwares so I brought it forward if it is not needed it should be removed from the other firmwares as well, let me know.